components/tools/OmeroJava/test/integration/DuplicationTest.java

```diff
@@ -265,15 +276,31 @@ private static void assertSameProperties(StatsInfo original, StatsInfo duplicate
         Assert.assertEquals(duplicate.getGlobalMax().getValue(), original.getGlobalMax().getValue());
     }
 
+    /**
+     * Assert that the given instances have the same property values.
+     * @param original the original instance
+     * @param duplicate the duplicate instance that is expected to correspond to the original
+     */
+    private static void assertSameProperties(Shape original, Shape duplicate) {
+      Assert.assertEquals(duplicate.getTheZ().getValue(), original.getTheZ().getValue());
+      Assert.assertEquals(duplicate.getTheT().getValue(), original.getTheT().getValue());
+      Assert.assertEquals(duplicate.getTheC().getValue(), original.getTheC().getValue());
+      ExternalInfo originalExternalInfo = original.getDetails().getExternalInfo();
+      Assert.assertNotNull(originalExternalInfo);
+      ExternalInfo duplicateExternalInfo = duplicate.getDetails().getExternalInfo();
+      Assert.assertNotNull(duplicateExternalInfo);
+      Assert.assertEquals(originalExternalInfo.getEntityId(), duplicateExternalInfo.getEntityId());
+      Assert.assertEquals(originalExternalInfo.getEntityType(), duplicateExternalInfo.getEntityType());
+      Assert.assertEquals(originalExternalInfo.getUuid(), duplicateExternalInfo.getUuid());
```

Comment:
To avoid confusion in case of a failure
The order should be the same: ``expected, actual``
```
Assert.assertEquals(duplicate.getTheC().getValue(), original.getTheC().getValue())
Assert.assertEquals(originalExternalInfo.getUuid(), duplicateExternalInfo.getUuid());
```

Reply:
Done in https://github.com/ome/openmicroscopy/pull/6334/commits/be66ef0f2f71470ce045024274f9c72ef4efd8e4. 

> The order should be the same: expected, actual

Speaking of confusion, this is the junit contract 😄. [testng](https://www.javadoc.io/doc/org.testng/testng/6.8.17/org/testng/Assert.html) expects `actual, expected` as the order (which is indeed `original, duplicate` in our case)